### PR TITLE
initial create of kiali (service mesh observability) and istio prompt refactoring

### PR DESCRIPTION
**Pull request by srinivasa-vasu**
- initial creation of kiali (kiali is for service mesh observability to understand the mesh composition and connectivity)
- refactor istio prompt (manual inject option is removed as auto inject is the default post 1.9)
- Envoy sidecars for microservices alone (db, elastic etc. are excluded)

### CODE REVIEW

File: generators/kubernetes/templates/kubectl-apply.sh.ejs [hunk 1]
```diff
@@ -55,3 +55,54 @@ kubectl apply -f monitoring/jhipster-prometheus-cr.yml
 kubectl apply -f monitoring/jhipster-grafana.yml
 kubectl apply -f monitoring/jhipster-grafana-dashboard.yml
 <%_ } _%>
+
```
Comment (PierreBesson) [inline generators/kubernetes/templates/kubectl-apply.sh.ejs]: I'm a bit reluctant to have so much complexity in the `kubectl-apply.sh` script. Can you extract all of this code a new `kiali-setup.sh` script.

**Comment by deepu105** [comment]
This would need a change in JHipster-core as well for JDL.

**Comment by DanielFran** [comment]
@srinivasa-vasu Can you please fix the conflicts?
@PierreBesson as soon as conflicts are fixed, it should be awesome to have it included for v6...

**Comment by DanielFran** [comment]
@MathieuAA Can you please take care of jhipster-core impacts?

**Comment by MathieuAA** [comment]
@DanielFran no prob, but what are they?

**Comment by DanielFran** [comment]
I believe @deepu105 was talking about the application configuration in jdl.

@srinivasa-vasu Can you please confirm if in terms of confuguration there is any changes?

**Comment by MathieuAA** [comment]
I just need the changes (this PR's so big I don't think I'll enjoy deducing what needs changing in JCore)

**Comment by DanielFran** [comment]
@MathieuAA From what I found:
istio: should be now true/false since autoinjection will be always available (options no, manualInjection, autoInjection are not needed anymore)
istioRoute: removed

**Comment by DanielFran** [comment]
@MathieuAA There was already one issue open by @deepu105 about those changes.
Will updated it with what I found...

**Comment by DanielFran** [comment]
It will be needed also to review documentation:
- https://www.jhipster.tech/jdl/#deploymentdeclaration
- https://www.jhipster.tech/jdl/#deployment_options

**Comment by deepu105** [comment]
@DanielFran @srinivasa-vasu can you guys merge it if ready

**Comment by DanielFran** [comment]
@srinivasa-vasu Thanks for this great enhancement.
Please do not forget to review the documentation.

**Comment by DanielFran** [comment]
Seems there is and issue with validations of expected files in generator tests. Will take a look...

**Comment by DanielFran** [comment]
@srinivasa-vasu I fixed the tests for Kubernetes Helm sub-generator.

Did you forget to implement same changes for Kubernetes Helm?

**Comment by srinivasa-vasu** [comment]
Will check.

**Comment by srinivasa-vasu** [comment]
@DanielFran Thnx for pulling-in the change.

**Pull request merged**